src/Npgsql/KerberosUsernameProvider.cs

```diff
@@ -17,22 +18,30 @@ class KerberosUsernameProvider
     static string? _principalWithoutRealm;
 
     internal static string? GetUsername(bool includeRealm, ILogger connectionLogger)
-    {
-        if (!_performedDetection)
-        {
-            DetectUsername(connectionLogger);
-            _performedDetection = true;
-        }
-        return includeRealm ? _principalWithRealm : _principalWithoutRealm;
-    }
+        => _performedDetection
+            ? includeRealm ? _principalWithRealm : _principalWithoutRealm
+#if NET5_0_OR_GREATER
+            : DetectUsername(includeRealm, connectionLogger, false).GetAwaiter().GetResult();
```

Comment:
Uh, do we want to do sync-over-async here? Shouldn't we just do sync on old NETs (or if async is false)?

Reply:
If you look properly there's no sync over async here (but there us one in the place @vonzshik caught, where I ignore the async flag).

Reply:
Ah right, my bad. It's helpful to add explicit `async: false` to make that slightly clearer.